driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -1607,16 +1821,129 @@ public ByteBuffer serialize(TupleValue value) {
+
+        public ParsingTypeCodec(Class<T> javaType) {
+            this(TypeToken.of(javaType));
+        }
+
+        public ParsingTypeCodec(TypeToken<T> javaType) {
+            this(VarcharCodec.instance, javaType);
+        }
+
+        public ParsingTypeCodec(StringCodec codec, Class<T> javaType) {
+            this(codec, TypeToken.of(javaType));
+        }
+
+        public ParsingTypeCodec(StringCodec codec, TypeToken<T> javaType) {
+            super(codec.getCqlType(), javaType);
+            this.codec = codec;
+        }
+
+        @Override
+        public ByteBuffer serialize(T value, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            return codec.serialize(format(value), protocolVersion);
+        }
+
+        @Override
+        public T deserialize(ByteBuffer bytes, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            return parse(codec.deserialize(bytes, protocolVersion));
+        }
+
+    }
+
+    /**
+     * An abstract TypeCodec that maps a Java Pojo to another Java object
+     * that can in turn be serialized into a CQL type.
+     * This can serve as a base for libraries dealing with Pojo mappings.
+     *
+     * @param <T> The outer Java type
+     * @param <U> The inner Java type
+     */
+    public abstract static class MappingCodec<T, U> extends TypeCodec<T> {
+
+        protected final TypeCodec<U> innerCodec;
+
+        public MappingCodec(TypeCodec<U> innerCodec, Class<T> javaType) {
+            this(innerCodec, TypeToken.of(javaType));
+        }
+
+        public MappingCodec(TypeCodec<U> innerCodec, TypeToken<T> javaType) {
+            super(innerCodec.getCqlType(), javaType);
+            this.innerCodec = innerCodec;
+        }
+
+        @Override
+        public ByteBuffer serialize(T value, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            return innerCodec.serialize(serialize(value), protocolVersion);
+        }
+
+        @Override
+        public T deserialize(ByteBuffer bytes, ProtocolVersion protocolVersion) throws InvalidTypeException {
+            return deserialize(innerCodec.deserialize(bytes, protocolVersion));
+        }
+
+        @Override
+        public T parse(String value) throws InvalidTypeException {
+            return deserialize(innerCodec.parse(value));
+        }
+
+        @Override
+        public String format(T value) throws InvalidTypeException {
+            return innerCodec.format(serialize(value));
+        }
+
+        protected abstract T deserialize(U value);
+
+        protected abstract U serialize(T value);
+
+    }
+
+    /**
+     * A codec that serializes Enum instances as CQL {@link DataType#varchar() varchar}s representing the Enum instance name
+     * as returned by {@link Enum#name()}.
+     *
+     * @param <E> The Enum class this codec serializes from and deserializes to.
+     */
+    public static class EnumCodec<E extends Enum<E>> extends ParsingTypeCodec<E> {
```

Comment:
I could also see the desire to map Enums to ordinal positioning.  (i.e. Days { Monday, Tuesday, Wendesday } would map Map Days.Monday to 0, Days.Tuesday to 1 and so on.